File: backend/src/main/java/com/sisvendas/sisvendas/service/ProdutoService.java
package com.sisvendas.sisvendas.service;

import org.springframework.stereotype.Service;
import com.sisvendas.sisvendas.models.Produto;
import com.sisvendas.sisvendas.repositories.IProdutoRepository;

@Service
public class ProdutoService 
    extends BaseService<Produto, IProdutoRepository> {
    
}

File: frontend/src/services/produto.service.ts
import { Injectable } from "@angular/core";
import { ProdutoModel } from "../models/produto.model";
import { BaseService } from "./base.service";
import { HttpClient } from "@angular/common/http";

@Injectable({
    providedIn: 'root'
})
export class ProdutoService extends BaseService<ProdutoModel> {
    protected override modulo: String = "produtos";

    constructor(httpClient: HttpClient) {
        super(httpClient);
    }
}